Go/Go.sublime-syntax

```diff
@@ -186,25 +190,67 @@ contexts:
-    # places that are "left over". Detecting redefinition is beyond the scope of
-    # this syntax engine.
+  #
+  # In Go, the predeclared constants are not keywords, and can be redefined. In
+  # many places such as variable declarations, types, function names, etc, we
+  # allow them to be scoped the same way as other identifiers. This "constant"
+  # rule should be used in places that are "left over". Detecting redefinition
+  # would be ideal, but is beyond the scope of this syntax engine; we simply
+  # expect it to be very rare.
+  match-predeclared-constants:
     - match: \b(?:true|false|nil)\b
       scope: constant.language.go
-    - include: match-new
-    - include: match-make
 
-  # Note: calls and casts are syntactically identical. Detecting casts and
-  # scoping them as types is beyond the capabilities of this syntax engine.
+  # Note: in Go, calls and casts are syntactically identical. Detecting casts
+  # and scoping them as types is beyond the capabilities of this syntax engine.
+  #
+  # https://golang.org/ref/spec#Predeclared_identifiers
   #
-  # This is somewhat duplicated in `match-selector`, which must be included
-  # before this rule.
+  # Notes on built-in functions
+  #
+  # Most built-in functions don't need special syntactic support. We scope them
+  # for the benefit of the users who prefer to distinguish them from
+  # user-defined identifiers. Two exceptions are `make` and `new`, where the
+  # first argument is scoped as a type, matching the special-case support in
+  # the compiler. When built-ins are redefined, this leads to incorrect
+  # scoping; like with constants, we expect such redefinition to be very rare.
+  #
+  # Note that we limit this detection to plain function calls, ignoring method
+  # calls and other identifier occurrences. The language currently allows
+  # built-in functions ONLY in a function call position. This helps minimize
+  # false positives.
+  #
+  # Notes on built-in types
+  #
+  # Unlike casts involving a user-defined type, casts involving a built-in
+  # types could be scoped purely as types rather than function calls. However,
+  # we stick to `variable.function.go` to make the treatment of built-ins
+  # purely additive, allowing the user to opt out.
   match-call-or-cast:
+    - match: \b(?:make|new)\b(?=(?:{{noise}}\))*{{noise}}\()
+      scope: variable.function.go support.function.builtin.go
+      push: pop-arguments-starting-with-type
+    - match: '{{predeclared_type}}(?=(?:{{noise}}\))*{{noise}}\()'
+      scope: variable.function.go support.type.builtin.go
+    - match: '{{predeclared_func}}(?=(?:{{noise}}\))*{{noise}}\()'
+      scope: variable.function.go support.function.builtin.go
+    - match: '{{ident}}(?=(?:{{noise}}\))*{{noise}}\()'
+      scope: variable.function.go
+
+  # See notes on `match-call-or-cast`.
+  pop-call-or-cast:
+    - match: \b(?:make|new)\b(?=(?:{{noise}}\))*{{noise}}\()
```

Comment:
`(?=(?:{{noise}}\))*` seems like a weird pattern. (occurs a few times)

Reply:
After double-checking, this seems to be as intended. The idea is to capture an identifier followed by zero or more `)` then by a `(`, ignoring any whitespace and comments in between. I don't see a simpler way.

Reply:
To clarify, this is valid Go:

```go
var _ = (((new)))(int)
```

Reply:
Nevermind, I missed a backslash there and though the `*` was applied to the look-ahead, but instead this is still within the look-ahead itself.